internal/controller/clustersecurityintentbinding_controller.go

```diff
@@ -59,7 +63,20 @@ func (r *ClusterSecurityIntentBindingReconciler) Reconcile(ctx context.Context,
 		logger.Info("ClusterSecurityIntentBinding configured", "ClusterSecurityIntentBinding.Name", req.Name)
 	}
 
-	if err = r.updateCsibStatus(ctx, logger, req); err != nil {
+	// Check if the object was previouly marked as invalid
+	if csib.Status.Status == StatusValidationFail {
+		logger.Info("ClusterSecurityIntentBinding found, not valid", "ClusterSecurityIntentBinding.Name", req.Name)
+		return doNotRequeue()
+	}
+
+	if csib.Status.Status == "" && !r.isValidCsib(ctx, logger, req) {
```

Comment:
For validating a CR, we should use either a [validating admission policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/) or a [validating admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook). Since validation after creation wouldn't prevent invalid CRs from being created in the first place.

Reply:
Agree. But that is a separate feature to be developed, and providing this stop-gap in the interim

Reply:
Then validation doesn't make sense.

Reply:
Disagree. It has value since the status will be shown as ValidationFail